Makefile

```diff
@@ -1,2 +1,151 @@
+
+update_barrier_test.o: test/update_barrier_test.c
+	$(CC) -g -c $(LIBFLAGS) test/update_barrier_test.c -o build/update_barrier_test.o
+
+update_player_pos_test.o: test/update_player_pos_test.c
+	$(CC) -g -c $(LIBFLAGS) test/update_player_pos_test.c -o build/update_player_pos_test.o
+
+update_selector_test.o: test/update_selector_test.c
+	$(CC) -g -c $(LIBFLAGS) test/update_selector_test.c -o build/update_selector_test.o
+	
+win_game_test.o: test/win_game_test.c
+	$(CC) -g -c $(LIBFLAGS) test/win_game_test.c -o build/win_game_test.o
+	
+
+
+
+#TARGET TO GENERATE THE EXECUTABLE OF THE PROGRAM (and the tests) - LINK OBJECT FILES
+
+tests: choose_menu_item_test.o game_over_test.o generate_view_test.o is_barrier_hit_test.o is_player_hit_test.o is_reach_goal_test.o load_map_test.o print_menu_test.o take_heart_test.o update_arrow_test.o update_barrier_test.o update_player_pos_test.o update_selector_test.o win_game_test.o
+
+	$(CC) -g -o bin/choose_menu_item_test build/choose_menu_item_test.o
+	$(CC) -g -o bin/game_over_test build/game_over_test.o
+	$(CC) -g -o bin/generate_view_test build/generate_view_test.o
+	$(CC) -g -o bin/is_barrier_hit_test build/is_barrier_hit_test.o
+	$(CC) -g -o bin/is_player_hit_test build/is_player_hit_test.o
+	$(CC) -g -o bin/is_reach_goal_test build/is_reach_goal_test.o
+	$(CC) -g -o bin/load_map_test build/load_map_test.o
+	$(CC) -g -o bin/print_menu_test build/print_menu_test.o
+	$(CC) -g -o bin/take_heart_test build/take_heart_test.o
+	$(CC) -g -o bin/update_arrow_test build/update_arrow_test.o
+	$(CC) -g -o bin/update_barrier_test build/update_barrier_test.o
+	$(CC) -g -o bin/update_player_pos_test build/update_player_pos_test.o
+	$(CC) -g -o bin/update_selector_test build/update_selector_test.o
+	$(CC) -g -o bin/win_game_test build/win_game_test.o
+
+
+	
+main: main.o credit.o game_over.o generate_view.o is_barrier_hit.o is_player_hit.o is_reach_goal.o load_map.o print_menu.o single_player.o take_heart.o update_arrow.o update_barrier.o update_player_pos.o user_manual.o win_game.o
+
+	$(CC) -g -o bin/Run_and_Shoot build/main.o
+	$(CC) -g -o bin/credit build/credit.o
+	$(CC) -g -o bin/game_over build/game_over.o
+	$(CC) -g -o bin/generate_view build/generate_view.o
+	$(CC) -g -o bin/is_barrier_hit build/is_barrier_hit.o
+	$(CC) -g -o bin/is_player_hit build/is_player_hit.o
+	$(CC) -g -o bin/is_reach_goal build/is_reach_goal.o
+	$(CC) -g -o bin/load_map build/load_map.o
+	$(CC) -g -o bin/print_menu build/print_menu.o
+	$(CC) -g -o bin/single_player single_player/main.o
+	$(CC) -g -o bin/take_heart build/take_heart.o
+	$(CC) -g -o bin/update_arrow build/update_arrow.o
+	$(CC) -g -o bin/update_barrier build/update_barrier.o
+	$(CC) -g -o bin/update_player_pos build/update_player_pos.o
+	$(CC) -g -o bin/user_manual build/user_manual.o
+	$(CC) -g -o bin/win_game build/win_game.o
+
+#TARGET TO GENERATE ALL THE EXECUTABLES (MAIN PROGRAM + TESTS TOGETHER)
+all: main tests
```

Comment:
I believe it should be the first rule to be executed, so put it on top of the file.

Reply:
Yes You are completely right. I was curious to see main results first. I forgot to change the order :D